File: perfbench-code/swicket-jpa/src/main/scala/swicketjpa/wicket/PasswordPage.scala
package swicketjpa.wicket

import org.apache.wicket.markup.html.form.Form
import org.apache.wicket.markup.html.form.PasswordTextField
import org.apache.wicket.markup.html.form.validation.EqualPasswordInputValidator
import org.apache.wicket.markup.html.link.BookmarkablePageLink
import org.apache.wicket.model.CompoundPropertyModel
import org.apache.wicket.model.Model

class PasswordPage extends TemplatePage {

  add(new Form("form") {
    val user = getBookingSession().user
    setDefaultModel(new CompoundPropertyModel(user))
    val passwordField = new PasswordTextField("password")
    add(new EditBorder("passwordBorder", passwordField))
    val verifyField = new PasswordTextField("verify", new Model(""))
    add(new EditBorder("verifyBorder", verifyField))
    add(new EqualPasswordInputValidator(passwordField, verifyField))
    add(new BookmarkablePageLink("cancel", classOf[MainPage]))

    override def onSubmit = {
      getEntityManager().merge(user)
      getSession().info("Password updated")
      setResponsePage(classOf[MainPage])
    }

  })

}

File: perfbench-code/wicket6-jpa/src/main/java/wicketjpa/wicket/PasswordPage.java
package wicketjpa.wicket;

import org.apache.wicket.markup.html.form.Form;
import org.apache.wicket.markup.html.form.FormComponent;
import org.apache.wicket.markup.html.form.PasswordTextField;
import org.apache.wicket.markup.html.form.validation.EqualPasswordInputValidator;
import org.apache.wicket.markup.html.link.BookmarkablePageLink;
import org.apache.wicket.model.CompoundPropertyModel;
import org.apache.wicket.model.Model;
import wicketjpa.entity.User;

public class PasswordPage extends TemplatePage {

    public PasswordPage() {
        add(new PasswordForm("form"));
    }

    private class PasswordForm extends Form<User> {

        private User user = getBookingSession().getUser();

        public PasswordForm(String id) {
            super(id);
            setModel(new CompoundPropertyModel<User>(user));
            FormComponent passwordField = new PasswordTextField("password");            
            add(new EditBorder("passwordBorder", passwordField));
            FormComponent verifyField = new PasswordTextField("verify", Model.of(""));
            add(new EditBorder("verifyBorder", verifyField));            
            add(new EqualPasswordInputValidator(passwordField, verifyField));            
            add(new BookmarkablePageLink<Void>("cancel", MainPage.class));
        }

        @Override
        protected void onSubmit() {            
            getEntityManager().merge(user);
            getSession().info("Password updated");
            setResponsePage(MainPage.class);
        }        
    }    
}
